### DISCUSSION TITLE
Is there a way to append an xarray data set?

### DISCUSSION THREAD

**Discussion by pecos27** [pydata/xarray#5124] [Q&A]
This question is based on one I asked yesterday: #5119. Apologies for double posting in this forum.

But now I am wondering if it's possible to append an xarray data set?

I have monthly data set in dimensions (time, depth, y, and x). and I am hoping to regroup the data into months (Jan - Dec), # of years in that month, depth, y, and x. Apologies in advance for not having reproducible data... just think of this data set as monthly time steps and spatial coordinates:

In my case, I have 480 monthly data points, which would equal for 12 months, there would be 40 years.

I tried:

```python
new_ds = []

for i in range(12): #12 months
    new_ds.append(ds.isel(time=slice(i,ds.shape[0],12)))  #ds.shape[0] is the size of the time axis
```

which outputs the right thing, where for each month there's 40 time points per month (January - December):

![Screenshot from 2021-04-07 11-57-36](https://user-images.githubusercontent.com/53262566/113849912-367f4400-979a-11eb-8846-e675b6be68e8.png)

However, I tried using `xr.concat` to try and put the new xarray dataframe `new_ds` into "one dataframe":

`combined = xr.concat(new_ds,dim='test')`

but now it outputs a xarray frame as shape: 12, 480, z, y, and x. Instead of what `new_ds` does.
Is the issue with my iterating and appending or with the concatetanation?

**Comment by RichardScottOZ** [comment]
What do you want it to look like?  You are making a list of 12 there, so the first dimension sounds right - and 480 is 12x40?

**Comment by dcherian** [comment]
This looks like a bug but it seems unlikely. 

It should be concatenating over a new dimension named `test` with size 40. Can you create a reproducible example please?
